packages/http/src/validator/deserializers/index.ts

```diff
@@ -18,4 +20,10 @@ export const query = new HttpParamDeserializerRegistry([
   new DeepObjectStyleDeserializer(),
 ]);
 
+export const path = new HttpParamDeserializerRegistry([
+  new SimpleStyleDeserializer(),
+  new LabelStyleDeserializer(),
+  new MatrixStyleDeserializer(),
+]);
```

Comment:
One thing to keep in mind is that we can in the future employ a library doing all this for us (https://github.com/litejs/uri-template-lite#readme) — this apparently passes all the URI Templates specification tests and has both a "match", and "expand" function we could use.

@StefanDywersant Would it be possible to do a quick (no more than 1H) test on how hard would it be to use this library instead of writing our deserialiser?

Reply:
There are two things making it a little harder to implement:
1) OpenApi's param serialization is a superset of what is inside URI Template RFC (deepobject, form, ..)
2) in order to perform extraction we'd need to generate URI Template - again, OpenApi does not help us here because OAS's `path` is not a URI Template.

Reply:
yeah all the csv supporting, split, matrix, pipes, etc is all OpenAPI-specific. This is not a complication of OpenAPI, more of a recognitiion of all the things people do in the wild which do not perfectly rely on standards. If everything was perfectly built using defined standards, our tooling would be easier!

Reply:
What did we do to deserve this? :trollface: — resolving :)